### ISSUE TITLE
browser.newPage freezes (never resolves)

### ISSUE THREAD

**Issue by pqvst** [puppeteer/puppeteer#4039]
### Steps to reproduce

**Tell us about your environment:**

* Puppeteer version: 1.12.2
* Platform / OS version: Ubuntu 16.04.5 & macOS
* Node.js version: 11.10

**What steps will reproduce the problem?**

Here's the exact code I'm using:

```
const browser = await puppeteer.launch({ 
  headless: true,
  args: ['--no-sandbox', '--disable-setuid-sandbox', '--lang=en-GB'],
});
const page = await browser.newPage();
```

With the following dockerfile:

```
FROM node:11

########################################################################################################################
# Puppeteer Begin
########################################################################################################################

# See https://crbug.com/795759
RUN apt-get update && apt-get install -yq libgconf-2-4

# Install latest chrome dev package and fonts to support major charsets (Chinese, Japanese, Arabic, Hebrew, Thai and a few others)
# Note: this installs the necessary libs to make the bundled version of Chromium that Puppeteer installs, work.
RUN apt-get update && apt-get install -y wget --no-install-recommends \
    && wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - \
    && sh -c 'echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list' \
    && apt-get update \
    && apt-get install -y google-chrome-unstable fonts-ipafont-gothic fonts-wqy-zenhei fonts-thai-tlwg fonts-kacst ttf-freefont \
      --no-install-recommends \
    && rm -rf /var/lib/apt/lists/* \
    && apt-get purge --auto-remove -y curl \
    && rm -rf /src/*.deb

########################################################################################################################
# Puppeteer End
########################################################################################################################

# Create app directory
WORKDIR /usr/src/app

# Install dependencies
COPY package.json .
COPY package-lock.json .
RUN npm install

# Bundle remaining sources
COPY . .

EXPOSE 3000

CMD [ "node", "./bin/www" ]
```

**What is the expected result?**

I expect the `browser.newPage` promise to *always* resolve or reject.

**What happens instead?**

After running this code for several hours with many iterations of the above code snippet, the `browser.newPage` promise will eventually not resolve nor reject causing my application to hang.

~~Note: Problem goes away when downgrading to puppeteer 1.11.0~~

Edit: Actually, downgrading to 1.11.0 did not help...

See older similar issue: https://github.com/GoogleChrome/puppeteer/issues/1409#issuecomment-464866096

**Comment by hg8** [comment]
Having the same problem I managed to fix it this way : 

```bash
# Install chromium separately
sudo apt install chromium-browser
# Install puppeteer CORE version 1.11.0
npm install puppeteer-core@1.11.0
```

Then in my code:
```js
const puppeteer = require('puppeteer-core');
const browser = await puppeteer.launch({
	executablePath: '/usr/bin/chromium-browser',
        headless: false,
      });
```

Hope this can be helpful.

**Comment by vongohren** [comment]
This is a comment that says how I solved the problem https://github.com/GoogleChrome/puppeteer/issues/1947#issuecomment-467112519

**Comment by justinmealey** [comment]
I solved the problem by setting the browser to use pipe instead of websocket:

`const browser = await puppeteer.launch({
     pipe: true
});`

**Comment by aslushnikov** [comment]
There's some expertise on running chromium in docker. We run our CI in docker, check out our troubleshooting on running in docker: [Running Puppeteer in Docker](https://github.com/GoogleChrome/puppeteer/blob/master/docs/troubleshooting.md#running-puppeteer-in-docker).

Hopefully either this or some other answers above will help! The Puppeteer - Docker integration falls outside of the scope of the project, so I'll close this.

**Comment by WasabiThumb** [comment]
> I solved the problem by setting the browser to use pipe instead of websocket:
> 
> `const browser = await puppeteer.launch({ pipe: true });`

Too bad that this fix leads to this happening instead
https://github.com/puppeteer/puppeteer/issues/1947

**Comment by WasabiThumb** [comment]
> > I solved the problem by setting the browser to use pipe instead of websocket:
> > `const browser = await puppeteer.launch({ pipe: true });`
> 
> Too bad that this fix leads to this happening instead
> #1947

More up-to-date version of, for practical purposes, the same issue https://github.com/puppeteer/puppeteer/issues/6258

**Comment by therepo90** [comment]
I made it work with some official docs and chatgpt

```
# Use the official Node.js 18 image
FROM node:18

# Set the working directory inside the container
WORKDIR /usr/src/app

# Copy package.json and package-lock.json first for better caching
COPY package*.json ./

# Install dependencies
RUN npm ci

# Install Chrome dependencies
RUN apt-get update && apt-get install -y wget gnupg libnss3 lsb-release ca-certificates  --no-install-recommends

# Add Google Chrome repository
RUN wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add -
RUN echo "deb [arch=amd64] http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list

# Install Google Chrome Stable
RUN apt-get update && apt-get install -y google-chrome-stable --no-install-recommends

# Install font packages
RUN apt-get install -y fonts-ipafont-gothic fonts-wqy-zenhei fonts-thai-tlwg fonts-kacst fonts-freefont-ttf libxss1 --no-install-recommends

# Clean up apt cache
RUN apt-get autoremove -y && apt-get clean && rm -rf /var/lib/apt/lists/*

# Add user so we don't need --no-sandbox.
RUN groupadd -r pptruser && useradd -r -g pptruser -G audio,video pptruser \
    && mkdir -p /home/pptruser/Downloads \
    && chown -R pptruser:pptruser /home/pptruser \
    && chown -R pptruser:pptruser /usr/src/app/node_modules \
    && chown -R pptruser:pptruser /usr/src/app/package.json \
    && chown -R pptruser:pptruser /usr/src/app/package-lock.json


# Copy the rest of the application code
COPY . .
RUN npm run build
EXPOSE 3000
CMD ["node", "server/dist/server/server.js" ]
```

**Comment by dlopel** [comment]
> This is a comment that says how I solved the problem [#1947 (comment)](https://github.com/puppeteer/puppeteer/issues/1947#issuecomment-467112519)

Following the comments solved my problem 😄 

puppeteer
```
const browser = await puppeteer.launch({
      headless: true,
      executablePath: process.env.CHROME_BIN || null,
      args: ['--no-sandbox', '--disable-setuid-sandbox', '--headless', '--disable-gpu', '--disable-dev-shm-usage'],
    });
```

dockerfile
```
FROM node:23-alpine AS dev
WORKDIR /home/base-api
ENV CHROME_BIN="/usr/bin/chromium-browser" \
    PUPPETEER_SKIP_CHROMIUM_DOWNLOAD="true"
RUN apk add --no-cache \
    git \
    udev \
    ttf-freefont \
    chromium
```

package.json
`"puppeteer": "^24.10.1",`
